src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs

```diff
@@ -79,7 +77,13 @@ public async Task StartAsync()
         /// <returns><c>true</c> if delay between next time we should ask for blocks is required; <c>false</c> otherwise.</returns>
         protected async Task<bool> SyncDepositsAsync()
         {
-            this.logger.LogInformation($"Fetching deposits from height {this.crossChainTransferStore.NextMatureDepositHeight}");
+            if (this.previousRequestedHeight == this.crossChainTransferStore.NextMatureDepositHeight)
```

Comment:
It is possible that the CCTS may not advanced and is still requesting the same deposits if:
- The CCTS is in suspended state. In this case simply check for suspended state and don't feed the CCTS any more.
- The CCTS could not process the deposit, possibly due to the source node being out of sync with other nodes. In this case allow the CCTS to re-request the same blocks as the deposits contained within may change favorably.